src/main/java/com/simibubi/create/content/trains/entity/Train.java

```diff
@@ -654,6 +714,12 @@ private void collideWithOtherTrains(Level level, Carriage carriage) {
 
 	public Pair<Train, Vec3> findCollidingTrain(Level level, Vec3 start, Vec3 end, ResourceKey<Level> dimension) {
 		Vec3 diff = end.subtract(start);
+		double length = diff.length();
+
+		if (length < 0.0001)
```

Comment:
Static final constant please, especially because you also use it here:
https://github.com/Creators-of-Create/Create/pull/9513/files#diff-671a8e994774f6be471b30fd8459587f56fd246119f83bb82b47b47e0056608cR1201